src/pages/home/report/ReactionList/BaseReactionList.tsx

```diff
@@ -1,64 +1,64 @@
 /* eslint-disable rulesdir/onyx-props-must-have-default */
 import Str from 'expensify-common/lib/str';
-import PropTypes from 'prop-types';
 import React from 'react';
 import {FlatList} from 'react-native';
+import type {FlatListProps} from 'react-native';
 import OptionRow from '@components/OptionRow';
-import participantPropTypes from '@components/participantPropTypes';
-import withWindowDimensions from '@components/withWindowDimensions';
 import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
 import Navigation from '@libs/Navigation/Navigation';
 import * as UserUtils from '@libs/UserUtils';
 import variables from '@styles/variables';
 import CONST from '@src/CONST';
 import ROUTES from '@src/ROUTES';
+import type {PersonalDetails} from '@src/types/onyx';
 import HeaderReactionList from './HeaderReactionList';
-import reactionPropTypes from './reactionPropTypes';
+import type ReactionListProps from './types';
 
-const propTypes = {
+type BaseReactionListProps = ReactionListProps & {
     /**
      *  Array of personal detail objects
      */
-    users: PropTypes.arrayOf(participantPropTypes).isRequired,
+    users: PersonalDetails[];
 
     /**
      * Returns true if the current account has reacted to the report action (with the given skin tone).
      */
-    hasUserReacted: PropTypes.bool,
+    hasUserReacted: boolean;
 
-    ...reactionPropTypes,
-};
-
-const defaultProps = {
-    hasUserReacted: false,
+    /**
+     * Returns true if the reaction list is visible
+     */
+    isVisible: boolean;
 };
 
 /**
  * Create a unique key for each action in the FlatList.
- * @param {Object} item
- * @param {Number} index
- * @return {String}
+ * @param item object
+ * @param index number
+ * @return string
  */
-const keyExtractor = (item, index) => `${item.login}+${index}`;
+const keyExtractor: FlatListProps<PersonalDetails>['keyExtractor'] = (item, index) => `${item.login}+${index}`;
 
 /**
  * This function will be used with FlatList getItemLayout property for optimization purpose that allows skipping
  * the measurement of dynamic content if we know the size (height or width) of items ahead of time.
  * Generate and return an object with properties length(height of each individual row),
  * offset(distance of the current row from the top of the FlatList), index(current row index)
  *
- * @param {*} _ FlatList item
- * @param {Number} index row index
- * @returns {Object}
+ * @param data FlatList item
+ * @param  index number - row index
+ * @returns object
  */
-const getItemLayout = (_, index) => ({
+const getItemLayout = (data: ArrayLike<PersonalDetails> | null | undefined, index: number): {length: number; offset: number; index: number} => ({
```

Comment:
Why `ArrayLike` is used here?

Reply:
@blazejkustra Hi. Thanks for your review. The reason behind this is that the `FlatList` component from React Native, which is later used in this file at line 109, requires the input of `getItemLayout` prop method to be an `ArrayLike<T>` which `T` is the type of the array that is getting passed as `data` prop, but as a single value. An example would be if the data is of type `TestType[]`, the `getItemLayout` method should get the first argument as `ArrayLike<TestType>`. This was just in the original docs of React Native. If you know a better way to achieve the same result, please let me know and I'll fix it eventually.
Thank.